File: Python/LongestConsecutiveSubsequence.py
class Solution:
    
    # arr[] : the input array
    # N : size of the array arr[]
    
    #Function to return length of longest subsequence of consecutive integers.
    def findLongestConseqSubseq(self,arr, N):
        present ={}
        for i in arr:
            present[i]=1
        checked ={}
        max_count = 0
        for n in arr:
            i = n
            if i-1 not in present.keys() and i not in checked.keys():
                count = 0
                while i in present.keys():
                    count+=1
                    checked[i]=1
                    max_count = max(max_count,count)
                    i+=1
        return max_count
                

File: Java/src/hashing/problems/LongestConsecutiveSubsequence.java
package hashing.problems;

// https://practice.geeksforgeeks.org/batch/cp-47/track/DSASP-Hashing/problem/longest-consecutive-subsequence2449
public class LongestConsecutiveSubsequence {

	static int findLongestConseqSubseq(int[] arr, int N) {
		if (arr.length == 0) return 0;
		int[] arr2 = new int[ 1000001 ];
		int count = 0;
		int maxInteger = -1;
		int maxLen = Integer.MIN_VALUE;
		for (int integer : arr) {
			arr2[ integer ]++;  // counting occurrence of each element and incrementing by 1
			maxInteger = Math.max ( maxInteger, integer );  // maxLen is the largest of array
		}
		for (int i = 0; i <= maxInteger; i++) {
			if (arr2[ i ] > 0) count++;    // increment count by 1 until arr2 contains arr[i+1]
			else {
				maxLen = Math.max ( maxLen, count );
				count = 0;
			}
			maxLen = Math.max ( maxLen, count );
		}
		return maxLen;
	}

	public static void main(String[] args) {
		System.out.println ( findLongestConseqSubseq ( new int[]{2, 6, 1, 9, 4, 5, 6, 3, 1000000}, 7 ) );
	}
}
